### PR TITLE
support append columns in create table as select

### PR DESCRIPTION
**Pull request by Junnplus**
Signed-off-by: ye.sijun <junnplus@gmail.com>

I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

```
mysql> create table foo (n int) ;
Query OK, 0 rows affected (0.01 sec)
Read 0 rows, 0 B in 0.003 sec., 0 rows/sec., 0 B/sec.

mysql> insert into foo(n) values(1);
Query OK, 0 rows affected (0.01 sec)
Read 1 rows, 4 B in 0.008 sec., 123.46 rows/sec., 493.86 B/sec.

mysql> CREATE TABLE bar (m INT) SELECT n FROM foo;
Query OK, 0 rows affected (0.05 sec)
Read 1 rows, 4 B in 0.044 sec., 22.64 rows/sec., 90.55 B/sec.

mysql> SELECT * FROM bar;
+------+------+
| m    | n    |
+------+------+
| NULL |    1 |
+------+------+
1 row in set (0.02 sec)
Read 1 rows, 9 B in 0.007 sec., 138.96 rows/sec., 1.25 KB/sec.
```

```
mysql> CREATE TABLE test1(a INT, b INT) ENGINE=MEMORY;                                                                                                                                                                               [45/1831]
Query OK, 0 rows affected (0.00 sec)
Read 0 rows, 0 B in 0.002 sec., 0 rows/sec., 0 B/sec.

mysql> insert into test1 VALUES (1, 2), (2, 3), (3, 4);
Query OK, 0 rows affected (0.00 sec)
Read 3 rows, 24 B in 0.001 sec., 2.62 thousand rows/sec., 20.92 KB/sec.

mysql> describe test1;
+-------+-------+------+
| Field | Type  | Null |
+-------+-------+------+
| a     | Int32 | YES  |
| b     | Int32 | YES  |
+-------+-------+------+
2 rows in set (0.01 sec)
Read 0 rows, 0 B in 0.003 sec., 0 rows/sec., 0 B/sec.

mysql> create table test2 select * from test1;
Query OK, 0 rows affected (0.03 sec)
Read 3 rows, 24 B in 0.026 sec., 116.82 rows/sec., 934.59 B/sec.

mysql> select * from test2;
+------+------+
| a    | b    |
+------+------+
|    1 |    2 |
|    2 |    3 |
|    3 |    4 |
+------+------+
3 rows in set (0.02 sec)
Read 3 rows, 24 B in 0.006 sec., 520.83 rows/sec., 4.17 KB/sec.

mysql> create table test3(x varchar, a varchar) select * from test1;
Query OK, 0 rows affected (0.07 sec)
Read 3 rows, 24 B in 0.066 sec., 45.32 rows/sec., 362.55 B/sec.

mysql> select * from test3;
+------+------+------+
| x    | a    | b    |
+------+------+------+
| NULL | 1    |    2 |
| NULL | 2    |    3 |
| NULL | 3    |    4 |
+------+------+------+
3 rows in set (0.01 sec)
Read 3 rows, 80 B in 0.007 sec., 430.28 rows/sec., 11.47 KB/sec.

mysql> describe test3;
+-------+--------+------+
| Field | Type   | Null |
+-------+--------+------+
| x     | String | YES  |
| a     | String | YES  |
| b     | Int32  | YES  |
+-------+--------+------+
3 rows in set (0.00 sec)
Read 0 rows, 0 B in 0.003 sec., 0 rows/sec., 0 B/sec.

mysql> create table test4(x varchar, a varchar) select b, a from test1;
Query OK, 0 rows affected (0.03 sec)
Read 3 rows, 24 B in 0.023 sec., 130.91 rows/sec., 1.05 KB/sec.

mysql> select * from test3;
+------+------+------+
| x    | a    | b    |
+------+------+------+
| NULL | 1    |    2 |
| NULL | 2    |    3 |
| NULL | 3    |    4 |
+------+------+------+
3 rows in set (0.02 sec)
Read 3 rows, 80 B in 0.007 sec., 433.77 rows/sec., 11.57 KB/sec.

mysql> create table test5(x varchar, a varchar) select b from test1;
Query OK, 0 rows affected (0.07 sec)
Read 3 rows, 12 B in 0.065 sec., 46.02 rows/sec., 184.09 B/sec.

mysql> select * from test5;
+------+------+------+
| x    | a    | b    |
+------+------+------+
| NULL | NULL |    2 |
| NULL | NULL |    3 |
| NULL | NULL |    4 |
+------+------+------+
3 rows in set (0.02 sec)
Read 3 rows, 78 B in 0.007 sec., 422.96 rows/sec., 11 KB/sec.

mysql> create table test6(x varchar, a varchar, b varchar) select a from test1;
Query OK, 0 rows affected (0.04 sec)
Read 3 rows, 12 B in 0.026 sec., 116.24 rows/sec., 464.97 B/sec.

mysql> select * from test6;
+------+------+------+
| x    | a    | b    |
+------+------+------+
| NULL | 1    | NULL |
| NULL | 2    | NULL |
| NULL | 3    | NULL |
+------+------+------+
3 rows in set (0.01 sec)
Read 3 rows, 101 B in 0.007 sec., 405.56 rows/sec., 13.65 KB/sec.
```
## Changelog

- Improvement

## Related Issues

Fixes [#2798](https://github.com/datafuselabs/databend/issues/3439)

## Test Plan

- [x] Unit Tests
- [x] Stateless Tests

### CODE REVIEW

**Comment by Junnplus** [comment]
Will also fix https://github.com/datafuselabs/databend/issues/3636

**Comment by BohuTANG** [comment]
> Will also fix #3636

Hello, will fix in this PR or later?

File: tests/suites/0_stateless/09_0004_remote_insert_into_select.result [hunk 1]
```diff
@@ -6,4 +6,6 @@
 99990000	100010000
 49995000	50005000
 NULL	0	2	c
+NULL	0	2	c
+NULL	0	4	c
```
Comment (Junnplus) [inline tests/suites/0_stateless/09_0004_remote_insert_into_select.result:10]: @BohuTANG Here, I think it has been fixed

**Review by dantengsky — APPROVED** [review]
LGTM

**Comment by BohuTANG** [comment]
Thank you @Junnplus , let's merge!

**Pull request merged**